Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -6,50 +6,667 @@
+
+    static void SetViewportUiClusterActiveButton(
+        AzToolsFramework::ViewportUi::ClusterId clusterId, AzToolsFramework::ViewportUi::ButtonId buttonId)
+    {
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::SetClusterActiveButton,
+            clusterId,
+            buttonId);
+    }
+
+    static AzToolsFramework::ViewportUi::ButtonId RegisterClusterButton(
+        AzToolsFramework::ViewportUi::ClusterId clusterId, const char* iconName)
+    {
+        AzToolsFramework::ViewportUi::ButtonId buttonId;
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::EventResult(
+            buttonId,
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::CreateClusterButton,
+            clusterId,
+            AZStd::string::format(":/stylesheet/img/UI20/toolbar/%s.svg", iconName));
+
+        return buttonId;
+    }
+
+    TransformMode::TransformMode(const AZ::EntityComponentIdPair& entityComponentIdPair)
+        : m_entityComponentIdPair(entityComponentIdPair)
+    {
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::EventResult(
+            m_transformClusterId,
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::CreateCluster,
+            AzToolsFramework::ViewportUi::Alignment::TopLeft);
+        m_transformTranslateButtonId = RegisterClusterButton(m_transformClusterId, "Move");
+        m_transformRotateButtonId = RegisterClusterButton(m_transformClusterId, "Rotate");
+        m_transformScaleButtonId = RegisterClusterButton(m_transformClusterId, "Scale");
+
+        // set translation tooltips
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::SetClusterButtonTooltip,
+            m_transformClusterId,
+            m_transformTranslateButtonId,
+            ManipulatorModeClusterTranslateTooltip);
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::SetClusterButtonTooltip,
+            m_transformClusterId,
+            m_transformRotateButtonId,
+            ManipulatorModeClusterRotateTooltip);
+        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(
+            AzToolsFramework::ViewportUi::DefaultViewportId,
+            &AzToolsFramework::ViewportUi::ViewportUiRequestBus::Events::SetClusterButtonTooltip,
+            m_transformClusterId,
+            m_transformScaleButtonId,
+            ManipulatorModeClusterScaleTooltip);
```

Comment:
FYI you can chain multiple calls to the same Bus Handler by using a lambda, since you have access to the Interface pointer.

Plus it is a more natural syntax since it is just several function calls

```suggestion

        AzToolsFramework::ViewportUi::ViewportUiRequestBus::Event(AzToolsFramework::ViewportUi::DefaultViewportId,
        [*this](AzToolsFramework::ViewportUi::ViewportUiRequests* requests)
        {
            m_transformClusterId = requests->CreateCluster(AzToolsFramework::ViewportUi::Alignment::TopLeft);
            m_transformTranslateButtonId = RegisterClusterButton(m_transformClusterId, "Move");
            m_transformRotateButtonId = RegisterClusterButton(m_transformClusterId, "Rotate");
            m_transformScaleButtonId = RegisterClusterButton(m_transformClusterId, "Scale");

            requests->SetClusterButtonTooltip(m_transformClusterId, m_transformTranslateButtonId, ManipulatorModeClusterTranslateTooltip);
            requests->SetClusterButtonTooltip(m_transformClusterId, m_transformRotateButtonId, ManipulatorModeClusterRotateTooltip);
            requests->SetClusterButtonTooltip(m_transformClusterId, m_transformScaleButtonId, ManipulatorModeClusterScaleTooltip);
        });
```